mlir/lib/Conversion/TosaToRock/TosaToRockPass.cpp

```diff
@@ -39,21 +40,25 @@ struct TosaToRock : public impl::TosaToRockPassBase<TosaToRock> {
     if (!func->hasAttr("kernel")) {
       return;
     }
+    // expand tosa.reshape to tensor.collapse_shape and tensor.expand_shape
+    // within kernel functions.
+    OpPassManager pm("func.func");
+    pm.addPass(mlir::tosa::createTosaToTensor());
```

Comment:
So that it only runs on `kernel` functions.

Reply:
Yes, this can be moved to the pipeline.

Reply:
I disagree with moving it to the pipeline, as I want this to only run on functions that are being `tosa-to-rock`d

Reply:
It was already in the pipeline, just not early enough. It is a normal conversion for tosa lowering (needed for the tosa.slice op).

Reply:
Few thoughts here : 

* I generally dont prefer passes buried within the passes.
* I thought tosa-to-rock was run on partitioned functions in the pipeline anyway.. but I can see the point of running the passes outside the driver.
* It also feels like we need a tensor-to-rock pass (maybe not right now)
   * it will also fit nicely with proposal of seperating tosa-to-tensor bits out of tosa-to-linalg and we will not have this somewhat weird depedency that tosa to needed to be converted to tensor before it can be converted to rock.

Reply:
Whereas I don't see the problem with nesting in a preprocessing pass that's only needed at this stage for tosa-to-rock conversion, and having it only run on the functions we're running tosa-to-rock on.

In the general case, tosa-to-tensor has to run _**after** tosa-to-linalg, because tosa-to-linalg can create a `tosa.reshape` now. However, we need those rewrites now - and only for functions targetting the GPU - in order to get the transforms thing working.

And since the only way to conditionally apply passes is to nest them, we need to nest them.

@manupak @sjw36  for "put it back"